packages/plugin-dev/src/browser/hosted-plugin-log-viewer.ts

```diff
@@ -17,8 +17,8 @@
 import { injectable, inject, postConstruct } from 'inversify';
 import { OutputChannel, OutputChannelManager } from '@theia/output/lib/common/output-channel';
 import { OutputContribution } from '@theia/output/lib/browser/output-contribution';
-import { HostedPluginWatcher } from './hosted-plugin-watcher';
-import { LogPart } from '../../common/types';
+import { LogPart } from '@theia/plugin-ext/lib/common/types';
```

Comment:
is that a third module will be there ?
because if we have
plugin-dev depends on plugin-ext it means we'll still have tons of dependencies coming

Reply:
Yes, `plugin-dev` depends on `plugin-ext`. I don't see any reason to make them independent.

Reply:
I think idea is to have `plugin-dev` as independent as possible so the bootstrap to run on each node could be minimalistic. (like avoiding any @theia/core dependency and a big set of transitive dependencies)

Reply:
I think using "plugin-dev" instead of "hosted" ends up with some weird words: "PluginDevPluginServer"...mayby "devmode"? DevmodePluginServer, DevmodePluginPreferences?

Reply:
Hm, I think you are miss understand intension of this PR. This is only first step for #3871. 
I will create separate PR for each step described there:  https://github.com/theia-ide/theia/issues/3871#issuecomment-489557854
So this PR only about extracting 'hosted mode'